swingset/src/main/java/com/nqadmin/swingset/SSBaseComboBox.java

```diff
@@ -111,33 +112,59 @@
 	private static final long serialVersionUID = 1L;
 
 	/**
-	 * Listener(s) for the component's value used to propagate changes back to bound
-	 * database column.
+	 * Listener(s) for the component's value used to propagate changes back to the rowset in certain instances.
+	 * <p>
+	 * NOTE: No guaranty whether ActionListener or FocusLister will be called first when focus is lost.
 	 */
 	protected class SSBaseComboBoxListener implements ActionListener, Serializable
 	{
 		private static final long serialVersionUID = -3131533966245488092L;
 
-		/** {@inheritDoc} */
+		/**
+		 * For JComboBox ActionListener and ItemListener are very similar.
+		 * 
+		 * {@inheritDoc} */
 		@Override
 		public void actionPerformed(final ActionEvent ae) {
 
-			removeRowSetListener();
-
-			M mapping = getSelectedMapping();
-
-			if (mapping == null) {
-				setBoundColumnText(null);
-				logger.debug(() -> String.format("%s: Setting to null.", getColumnForLog()));
-			} else {
-				setBoundColumnText(String.valueOf(mapping));
-				logger.debug(String.format("%s: Setting to %s.",  getColumnForLog(), mapping));
+			// If this is a combo navigator, SSSyncManager will have it's own listeners.
+			// This is just for keeping a bound column in sync.
+			//
+			// Could be combined with next block, but keeping them separate for debugging.
+			// May be able to eliminate actionListenerNoUpdate when GlazedLists fully supports STRICT/CONTAINS
+			if (getBoundColumnName() == null) {
+				logger.debug("{}: Action Listener returning. No bound column.", () -> getColumnForLog());
+				return;
 			}
-
-			addRowSetListener();
+			
+			// Could be combined with prior block, but keeping them separate for debugging.
+			// May be able to eliminate actionListenerNoUpdate when GlazedLists fully supports STRICT/CONTAINS
+			if (actionListenerNoUpdate) {
+				logger.debug("{}: Action Listener returning. actionListenerNoUpdate set to TRUE.", () -> getColumnForLog());
+				return;
+			}
```

Comment:
actionListenerNoUpdate is likely STRICT/CONTAINS  workaround specific. Presumably if GL fully supported it, we'd only ever have null or an SSListItem passed to setSelectedItem(). Will have to put some thought into when we'd expect null (insert row, some escaping from/clearing of the combo editor)? Maybe we'd only ever have null for the insert row OR getModel().getSize()==0 ?